test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -193,14 +188,19 @@ void performFailoverTest() throws InterruptedException {
                     } else {
                         log.debug("Read timeout");
                     }
-                } catch (ConnectionClosedException e) {
+                } catch (RetriesExhaustedException e) {
                     log.warn("Test exception in reading events: ", e);
                     continue;
-                } catch (ReinitializationRequiredException e) {
+                } catch (Throwable e) {
+                    if (e.getCause() instanceof RetriesExhaustedException) {
```

Comment:
Observing RetriesExhaustedException wrapped inside a RuntimeException
Created an issue for the same `https://github.com/pravega/pravega/issues/1687`